### PR TITLE
jsonpb.Parse with better error reporting

### PR DESCRIPTION
**Pull request by mandarjog**
This PR implements better error reporting for descriptors.
It ensures that adapters.Params, aspects.Params and all descriptors are parsed in the same way.

Errors are reported with a context.
The following errors says that the problem is in the 0th manifest, 6th attribute
```
* manifests[0].attributes[6]: unknown value "TIMESTAMP2" for enum istio.mixer.v1.config.descriptor.ValueType: [{"name":"request.time","value_type":"TIMESTAMP2"}],
```

Note: re cache_test.go
My PR build was failing, so I had to update this test to be more resilient. 

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/585)

### CODE REVIEW

File: pkg/config/descriptor/descriptor.go [hunk 1]
```diff
@@ -40,6 +53,18 @@ type Finder interface {
 	GetQuota(name string) *dpb.QuotaDescriptor
 }
 
+type dname string
+
+const (
```
Comment (douglas-reid) [inline pkg/config/descriptor/descriptor.go]: question: do we need a note somewhere about updating/maintaining this list?

File: pkg/config/descriptor/descriptor.go [hunk 2]
```diff
@@ -49,6 +74,144 @@ type finder struct {
 	attributes         map[string]*dpb.AttributeDescriptor
 }
 
+// typeMap maps descriptor types to example messages of those types.
+// Add example descriptors for all here instead of empty ones.
```
Comment (douglas-reid) [inline pkg/config/descriptor/descriptor.go]: should this be a TODO?  Do we want to go ahead and do this now?

File: pkg/config/manager.go [hunk 3]
```diff
@@ -179,13 +182,13 @@ func (c *Manager) fetchAndNotify() {
...
 		c.Unlock()
-		glog.Warningf("Error installing new config %s", err)
+		glog.Warningf("Error loading new config %v", err)
 	}
 	if rt == nil {
 		return
 	}
 
-	glog.Infof("Installing new config from %s ", c.store)
+	glog.Infof("Loaded new config %s ", c.store)
```
Comment (douglas-reid) [inline pkg/config/manager.go]: nit: any reason for the trailing space?

File: pkg/config/runtime.go [hunk 4]
```diff
@@ -123,6 +123,14 @@ const resolveSize = 50
 func resolve(bag attribute.Bag, kindSet KindSet, rules map[rulesKey]*pb.ServiceConfig, resolveRules resolveRulesFunc,
 	onlyEmptySelectors bool, identityAttribute string, identityAttributeDomain string, strictSelectorEval bool) (dlist []*pb.Combined, err error) {
 	scopes := make([]string, 0, 10)
+	if glog.V(2) {
+		pfx := ""
+		if onlyEmptySelectors {
+			pfx = "unconditionally "
+		}
+		glog.Infof("%sresolving for kinds: %s", pfx, kindSet)
```
Comment (douglas-reid) [inline pkg/config/runtime.go]: isn't this a potential dupe of the message on lines 105?  should we remove 105?

File: pkg/config/validator.go [hunk 5]
```diff
@@ -211,8 +211,8 @@ func (a adapterKey) String() string {
 // compatfilterConfig
 // given a yaml file, filter specific keys from it
 // globalConfig contains descriptors and adapters which will be split shortly.
-func compatfilterConfig(cfg string, shouldSelect func(string) bool) (data []byte, err error) {
-	var m = map[string]interface{}{}
+func compatfilterConfig(cfg string, shouldSelect func(string) bool) (data []byte, m map[string]interface{}, err error) {
+	m = map[string]interface{}{}
```
Comment (douglas-reid) [inline pkg/config/validator.go]: why do we take the map only to set it to empty?

File: testdata/configroot/scopes/global/adapters.yml [hunk 6]
```diff
@@ -8,7 +8,7 @@ adapters:
   - name: default
     impl: stdioLogger
     params:
-      logStream: 0 # STDERR
+      logStream: STDERR # STDERR
```
Comment (douglas-reid) [inline testdata/configroot/scopes/global/adapters.yml]: we can kill the comment now.

File: pkg/config/descriptor/descriptor.go (hunk 1 above)
Comment (mandarjog) [inline pkg/config/descriptor/descriptor.go]: Added note.

File: pkg/config/descriptor/descriptor.go (hunk 2 above)
Comment (mandarjog) [inline pkg/config/descriptor/descriptor.go]: Added examples of descriptors we use.

File: pkg/config/manager.go (hunk 3 above)
Comment (mandarjog) [inline pkg/config/manager.go]: fixed.

File: pkg/config/runtime.go (hunk 4 above)
Comment (mandarjog) [inline pkg/config/runtime.go]: removed.__

File: pkg/config/validator.go (hunk 5 above)
Comment (mandarjog) [inline pkg/config/validator.go]: fixed.

File: testdata/configroot/scopes/global/adapters.yml (hunk 6 above)
Comment (mandarjog) [inline testdata/configroot/scopes/global/adapters.yml]: done.

File: pkg/config/descriptor/descriptor.go [hunk 7]
```diff
@@ -49,6 +75,167 @@ type finder struct {
...
+	var err error
+
+	if enc, err = json.Marshal(obj); err != nil {
+		return ce.Append(ctx, err)
+	}
+	um := &jsonpb.Unmarshaler{AllowUnknownFields: allowUnknownFields}
+	if err = um.Unmarshal(bytes.NewReader(enc), dm); err != nil {
+		msg := fmt.Sprintf("%v: [%s]", err, string(enc))
+		if example != nil {
+			um := &jsonpb.Marshaler{}
```
Comment (douglas-reid) [inline pkg/config/descriptor/descriptor.go]: did you intentionally reuse `um` here?  `um` is earlier defined to be a `jsonpb.Unmarshaler. Might be clearer to call it jsonm or something.

**Pull request merged**